pkg/apis/security/v1alpha1/types.go

```diff
@@ -135,9 +135,17 @@ type NetworkPolicyPeer struct {
 	// workloads in To/From fields. If set with PodSelector,
 	// Pods are matched from Namespaces matched by the NamespaceSelector.
 	// Cannot be set with any other selector except PodSelector or
-	// ExternalEntitySelector.
+	// ExternalEntitySelector. Cannot be set with Namespaces.
 	// +optional
 	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// Select Pod/ExternalEntity from Namespaces matched by Namespaces.Selector,
+	// or from the same Namespace of the appliedTo workloads.
+	// Cannot be set with any other selector except PodSelector or
+	// ExternalEntitySelector. This field can only be set when NetworkPolicyPeer
+	// is created for ClusterNetworkPolicy ingress/egress rules.
+	// Cannot be set with NamespaceSelector.
+	// +optional
+	Namespaces *PeerNamespaces `json:"namespaces,omitempty"`
```

Comment:
This particular option has not been discussed I believe, and I agree it might make upgrade a bit easier. For upstream ClusterNetworkPolicy proposal it does not make a big difference because those are new API types to be added anyways. If everyone like the `SelfNamespace: bool` approach, I can make the change for this PR and propose the same thing upstream. I think it's perfectly okay if the upstream spec came out a bit different than what we merge in Antrea as well

Reply:
Updated the PR based on k8s-sig-network upstream KEP comments. Instead of a `bool` field, changed the `Self` keyword to a `NamespaceMatchType` for it to be more extensible, as well as keeping the old namespaceSelector semantics exactly as-is.